client/my-sites/backup/backup-storage-space/backup-storage-space-upsell/index.tsx

```diff
@@ -0,0 +1,99 @@
+/**
+ * External dependencies
+ */
+import React, { FunctionComponent, useCallback, useEffect } from 'react';
+import { Button } from '@wordpress/components';
+import { sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Gridicon from 'calypso/components/gridicon';
+import { recordTracksEvent } from 'calypso/state/analytics/actions/record';
+import { useTranslate, translate as translateType } from 'i18n-calypso';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+export type BackupStorageSpaceUpsellOptions =
+	| 'no_upsell'
+	| 'first_upsell'
+	| 'second_upsell'
+	| 'out_of_storage';
+
+const getStatusText = (
+	upsellOption: BackupStorageSpaceUpsellOptions,
+	storageLimit: number,
+	translate: typeof translateType
+) => {
+	switch ( upsellOption ) {
+		case 'first_upsell':
+			// TODO: calculate storage time, account for GB, and translate once API data is available.
+			return sprintf( 'You will reach your %1$sGB storage limit in %2$s days', storageLimit, 3 );
```

Comment:
For this case I didn't want to use `translate` just yet, since it will send the string off to be translated. I'm not sure how this string will be in the end, since the API might return something like "200GB" and "200MB", or "200" and "0.2", or "200000" and "200". I figured it is best to wait for that API data before making a string that will go for translation.